### ISSUE TITLE
namebench error 2: browser is not opened if user has not set a default web browser

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jeevandongre/namebench#43]
```
What steps will reproduce the problem?
1.Run namebench.exe v1.0.1.0 on Windows 7 x86


Initial unzip dialog does not close and hangs during benchmark. Upon saving 
CVS file following error dialogs appear:

[Error 2] the system cannot find the path specified

File "libnamebench\better_webbrowser.pyc", line 77, in open

File "webbrowser.pyc", line 61, in open

File "libnamebench\better_webbrowser.pyc", line 33, in 
create_win32_http_cmd

Along with a C++ runtime error in 
C:\users\username\appdata\local\temp\ckz_4O5G\namebench.exe

The relevent logfile shows:

Traceback (most recent call last):
  File "libnamebench\tk.pyc", line 87, in run
  File "libnamebench\base_ui.pyc", line 92, in RunBenchmark
  File "libnamebench\base_ui.pyc", line 111, in DisplayHtmlReport
  File "libnamebench\better_webbrowser.pyc", line 77, in open
  File "webbrowser.pyc", line 61, in open
  File "libnamebench\better_webbrowser.pyc", line 58, in open
  File "libnamebench\better_webbrowser.pyc", line 33, in 
create_win32_http_cmd
WindowsError: [Error 2] The system cannot find the file specified



```

Original issue reported on code.google.com by `n8la...@gmail.com` on 8 Dec 2009 at 2:38

Attachments:
- [namebench.JPG](https://storage.googleapis.com/google-code-attachments/namebench/issue-43/comment-0/namebench.JPG)

**Comment by GoogleCodeExporter** [comment]
```
I also am experiencing the same issue on Windows Vista.  Happens on completions 
of
test and producing of results.  it is repeatable.
```

Original comment by `inthe...@gmail.com` on 8 Dec 2009 at 2:50

**Comment by GoogleCodeExporter** [comment]
```
Same behavior observed on x86 WinXP SP3.
```

Original comment by `JLBengh...@gmail.com` on 8 Dec 2009 at 2:51

**Comment by GoogleCodeExporter** [comment]
```
i too got this error. it is repeatable.
```

Original comment by `jfd....@gmail.com` on 8 Dec 2009 at 3:22

**Comment by GoogleCodeExporter** [comment]
```
This issue occurs if the user has never set a default web browser, and has been 
fixed as 
part of revision 427. This fix makes it so that it should fall back to the 
system HTML 
handler, and will be included in 1.0.2, which should be released within the 
next day. 
Thanks!

```

Original comment by `thomas.r...@gmail.com` on 8 Dec 2009 at 3:45
- Changed title: **namebench error 2: browser is not opened if user has not set a default web browser**
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
Issue 46 has been merged into this issue.
```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 5:23

**Comment by GoogleCodeExporter** [comment]
```
What if the user HAS set a default browser? I use Opera 10 as my primary web 
browser, but IE 8 as the system default.
```

Original comment by `SILGoo...@gmail.com` on 8 Dec 2009 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
Issue 48 has been merged into this issue.
```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 7:32

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed as of version 1.0.2. It shouldl now properly fall back to 
the 
HKEY_LOCAL_MACHINE key, and then the HTML handler.

```

Original comment by `tstromb...@google.com` on 8 Dec 2009 at 7:34
- Changed state: **Fixed**
